xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-xwiki/src/main/java/org/xwiki/rendering/internal/resolver/AbstractRelativeResourceReferenceEntityReferenceResolver.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.internal.resolver;
+
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.rendering.listener.reference.ResourceReference;
+import org.xwiki.rendering.listener.reference.ResourceType;
+
+/**
+ * Abstract class for all relative resource reference resolvers.
+ *
+ * @version $Id$
+ * @since 17.0.0RC1
+ */
+public abstract class AbstractRelativeResourceReferenceEntityReferenceResolver
+    extends AbstractResourceReferenceEntityReferenceResolver
+{
+    @Inject
+    @Named("relative")
+    protected EntityReferenceResolver<String> relativeReferenceResolver;
+
+    /**
+     * @param type the resource type that this resolver will support
+     */
+    protected AbstractRelativeResourceReferenceEntityReferenceResolver(ResourceType type)
+    {
+        super(type);
+    }
+
+    @Override
+    public EntityReference resolve(ResourceReference resourceReference, EntityType entityType, Object... parameters)
```

Comment:
This seems to be almost an exact duplicate of AbstractResourceReferenceEntityReferenceResolver. I'm sure code reuse can be improved.